mmdet/apis/inference.py

```diff
@@ -131,21 +142,20 @@ def inference_detector(
             data_ = dict(img_path=img, img_id=0)
         # build the data pipeline
         data_ = test_pipeline(data_)
-        data.append(data_)
 
-    for m in model.modules():
-        assert not isinstance(
-            m,
-            RoIPool), 'CPU inference with RoIPool is not supported currently.'
+        data_['inputs'] = [data_['inputs']]
+        data_['data_samples'] = [data_['data_samples']]
+
+        # forward the model
+        with torch.no_grad():
+            results = model.test_step(data_)[0]
```

Comment:
Why not infer a batch of data and then `result_list.extend(results)`?

Reply:
Because if the batch is large, it will be OOM. If batch inference is supported, it needs to be combined with a split batch operation.